Bookish/Services/DatabaseSearch.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Reflection;
+
+namespace Bookish.Models;
+
+public abstract class DatabaseSearch
+{
+    // Tests if <searchParameter> is contained in <property> of <model> 
+    private static bool ModelHasMatchingProperty<TModel>(TModel model, PropertyInfo property, string searchParameter)
+    {
+        var searchTerms = searchParameter.Split(' ');
+        var modelPropertyValue = property.GetValue(model);
+        if (modelPropertyValue == null)
+        {
+            return false;
+        }
+        var modelPropertyWords = ((string) modelPropertyValue).ToLower().Split();
+        return searchTerms.All(word => modelPropertyWords.Contains(word.ToLower()));
+    }
+
+    protected static List<TModel> SearchThroughTable<TModel>(TModel searchParameters, string orderBy, List<TModel> table)
+    {
+        var orderByProperty = typeof(TModel).GetProperty(orderBy);
+        foreach (var property in typeof(TModel).GetProperties())
+        {
+            var parameter = property.GetValue(searchParameters);
+            if (property.Name != "Id" && parameter != null)

```

Comment:
This will go wrong for properties with non-null defaults like `int` and `DateTime`